pkg/sign/signer.go

```diff
@@ -15,29 +15,218 @@
 package sign
 
 import (
+	"bytes"
 	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
 	"crypto/rand"
 	"crypto/sha256"
 	_ "crypto/sha512" // if user chooses SHA2-384 or SHA2-512 for hash
 	"crypto/x509"
 	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
 	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
 
 	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
 	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
 )
 
 type Signer interface {
-	Sign(data []byte) (*protobundle.Bundle, error)
+	Sign(content Content) (*protobundle.Bundle, error)
 }
 
-// TODO: implement
-// type Fulcio struct {
-//	baseURL       string
-//	identityToken string
-// }
+type Fulcio struct {
+	options *FulcioOptions
+}
+
+type FulcioOptions struct {
+	// URL of Fulcio instance
+	BaseURL string
+	// OIDC token to send to Fulcio instance
+	IdentityToken string
+	// Optional timeout for network requests
+	Timeout time.Duration
+	// Optional version string for user agent
+	LibraryVersion string
+}
+
+func SignerFulcio(opts *FulcioOptions) *Fulcio {
```

Comment:
nit: `NewFulcioClient`?

Reply:
I ended up going with `NewFulcio`, to not confuse this with something that's just focused on HTTP communication with Fulcio.